client/src/pages/Course/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from "react";
+import { RouteComponentProps } from "react-router";
+import { getCourse, CourseData } from "utils/courses";
+import { getPosts, PostDataId } from "utils/posts";
+import { Button, Typography } from "@material-ui/core";
+import Navbar from "components/Navbar";
+import Post from "pages/Post";
+import NewPost from "pages/NewPost";
+
+const Course: React.FunctionComponent<RouteComponentProps> = ({
+  location: { search },
+}) => {
+  const [course, setCourse] = useState<CourseData>();
+  const [posts, setPosts] = useState<PostDataId[]>([]);
+  const [addingPost, setAddingPost] = useState(false);
+  const courseId = search.substring(1);
+
+  const getCourseInfo = async () => {
+    // TODO: Add error handling
+    const courseData = (await getCourse(courseId)) as CourseData;
+    setCourse(courseData);
+
+    const postsData = (await getPosts(courseId)) as PostDataId[];
+    setPosts(postsData);
+  };
+
+  const handleNewPost = () => {
+    setAddingPost(true);
+  };
+
+  const toggleNewPost = (exit: boolean) => {
+    setAddingPost(exit);
+  };
+
+  // TODO: Find a better way to do this
```

Comment:
Good point, I agree, since posts have sub-collections that are also updating, we would lose information if we manually added the new post to the state every time. We can revisit this idea later